File: Selection statement/Question3/question3.java
import java.util.Scanner;

public class question3 {
    static void StudentDiscount(int n) {
        float discount;

        if (n <= 500) {
            discount = (5 * n) / 100;
            System.out.println("Your discount amount is: " + discount);
        } else {
            discount = (10 * n) / 100;
            System.out.println("Your discount amount is: " + discount);
        }
    }

    static void NonStudentDiscount(int n) {
        float discount;

        if (n <= 500) {
            discount = (2 * n) / 100;
            System.out.println("Your discount amount is: " + discount);
        } else {
            discount = (8 * n) / 100;
            System.out.println("Your discount amount is: " + discount);
        }
    }

    public static void main(String[] args) {

        try (Scanner scan = new Scanner(System.in)) {
            char response;

            System.out.println("Are you a student? (y/n)");
            response = scan.next().charAt(0);

            if (response == 'Y' || response == 'y') {
                System.out.print("Enter your bill amount: ");
                int bill = scan.nextInt();
                StudentDiscount(bill);
            } else if (response == 'N' || response == 'n') {
                System.out.print("Enter your bill amount: ");
                int bill = scan.nextInt();
                NonStudentDiscount(bill);
            } else {
                System.out.println("Invalid input");
            }
        }
    }
}

File: Selection statement/Question3/question3.c
#include <stdio.h>

int StudentDiscount(int n)
{
    float discount;

    if (n <= 500)
    {
        discount = (5 / n) * 100;
        printf("Discount is %d\n", discount);
    }
    else
    {
        discount = (10 / n) * 100;
        printf("Discount is %d\n", discount);
    }
}

int NonStudentDiscount(int n)
{
    float discount;

    if (n <= 500)
    {
        discount = (2 / n) * 100;
        printf("Discount is %d\n", discount);
    }
    else
    {
        discount = (8 / n) * 100;
        printf("Discount is %d\n", discount);
    }
}

int main()
{
    char response;

    printf("Are you a student? (y/n)\n");
    scanf(" %c", &response);

    if (response == 'Y' || response == 'y')
    {
        int n;
        printf("Enter the bill amount: \n");
        scanf("%d", &n);
        StudentDiscount(n);
    }
    else if (response == 'N' || response == 'n')
    {
        int n;
        printf("Enter the bill amount: \n");
        scanf("%d", &n);
        NonStudentDiscount(n);
    }
    else
    {
        printf("Invalid input\n");
    }

    return 0;
}